src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -958,14 +975,18 @@ private void DeadlineExceededCallback(object? state)
                         DeadlineExceeded();
                         return;
                     }
-                }
 
-                // Deadline has not been reached because timer maximum due time was smaller than deadline.
-                // Reschedule DeadlineExceeded again until deadline has been exceeded.
-                GrpcCallLog.DeadlineTimerRescheduled(Logger, remaining);
+                    if (_deadlineTimer != null)
+                    {
+                        // Deadline has not been reached because timer maximum due time was smaller than deadline.
+                        // Reschedule DeadlineExceeded again until deadline has been exceeded.
+                        GrpcCallLog.DeadlineTimerRescheduled(Logger, remaining);
+
+                        var dueTime = CommonGrpcProtocolHelpers.GetTimerDueTime(remaining, Channel.MaxTimerDueTime);
+                        _deadlineTimer?.Change(dueTime, Timeout.Infinite);
```

Comment:
```suggestion
                        _deadlineTimer!.Change(dueTime, Timeout.Infinite);
```

Is `_deadlineTimer` still guaranteed to be set here?